src/Functions/map.cpp

```diff
@@ -147,6 +149,99 @@ class FunctionMap : public IFunction
     }
 };
 
+// mapFromArrays(keys, values) is a function that allows you to make key-value pair from a pair of arrays
+class FunctionMapFromArrays : public IFunction
+{
+public:
+    static constexpr auto name = "mapFromArrays";
+
+    static FunctionPtr create(ContextPtr)
+    {
+        return std::make_shared<FunctionMapFromArrays>();
+    }
+
+    String getName() const override
+    {
+        return name;
+    }
+
+    size_t getNumberOfArguments() const override
+    {
+        return 2;
+    }
+
+    bool isSuitableForShortCircuitArgumentsExecution(const DataTypesWithConstInfo & /*arguments*/) const override { return false; }
+    bool useDefaultImplementationForNulls() const override { return false; }
+    bool useDefaultImplementationForConstants() const override { return true; }
+
+    DataTypePtr getReturnTypeImpl(const DataTypes & arguments) const override
+    {
+        if (arguments.size() != 2)
+            throw Exception(ErrorCodes::NUMBER_OF_ARGUMENTS_DOESNT_MATCH,
+                "Function {} requires 2 arguments, but {} given", getName(), arguments.size());
+
+        const auto * keys_type = checkAndGetDataType<DataTypeArray>(arguments[0].get());
+        if (!keys_type)
+            throw Exception{"First argument for function " + getName() + " must be a Array",
+                            ErrorCodes::ILLEGAL_TYPE_OF_ARGUMENT};
+
+        const auto * values_type = checkAndGetDataType<DataTypeArray>(arguments[1].get());
+        if (!values_type)
+            throw Exception{"Second argument for function " + getName() + " must be a Array",
+                            ErrorCodes::ILLEGAL_TYPE_OF_ARGUMENT};
+
+        DataTypes tmp;
+        const auto & key_type = keys_type->getNestedType();
```

Comment:
Looking at [docs for map()](https://clickhouse.com/docs/en/sql-reference/functions/tuple-map-functions#map), the keys of a map can only be of reasonable data type such as String, Integer, LowCardinality, FixedString, UUID, Date, DateTime, Date32, Enum. Want to implement the same restriction here as well + also document it?